hack/test/e2e-iso.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+set -eou pipefail
+
+source ./hack/test/e2e.sh
+
+function cleanup() {
+  kill $(cat ${TMP}/qemu.pid); kill $(jobs -p) || true; rm -rf ${TMP}
+}
+
+mkdir -p ${TMP}
+
+qemu-img create -f qcow2 ${TMP}/disk.qcow2 4G
+
+qemu-system-x86_64 \
+  -m 2048 \
+  -curses \
+  -nographic \
+  -serial file:${TMP}/test.log \
+  -pidfile ${TMP}/qemu.pid \
+  -enable-kvm \
+  -cpu host \
+  -boot cd \
+  -netdev user,id=talos,ipv4=on,net=10.254.0.0/24,dhcpstart=10.254.0.10,hostfwd=tcp::50000-:50000,hostfwd=tcp::6443-:6443,hostname=control-plane-1 \
+  -device virtio-net,netdev=talos \
+  -cdrom ${ARTIFACTS}/talos-amd64.iso \
+  -hda ${TMP}/disk.qcow2 &
+
+trap 'cleanup' EXIT
+
+${TALOSCTL} gen config \
+  --additional-sans 127.0.0.1 \
+  --install-disk /dev/sda \
+  --install-image ghcr.io/talos-systems/installer:latest \
+  --output-dir ${TMP} \
+  test https://10.254.0.10:6443
+
+timeout -v --preserve-status 1m  bash -c "until ${TALOSCTL} apply-config -n 127.0.0.5 --insecure -f ${TMP}/init.yaml; do sleep 5; done"
```

Comment:
Should we also wait for apply config to finish and do something super simple as talosctl version with generated talosconfig? This would also test apply-config in maintenance mode

Reply:
Unfortunately we can't since the Talos certificates are not valid for `127.0.0.1`. I tried this. I added `--additional-cert-sans` but that only applies to Kubernetes.

Reply:
The good news is that we test apply-config so we aren't completely untested. Still would be good to get to one day, but changing the functionality of that flag felt out of scope and like a last minute design decision we shouldn't rush.

Reply:
I wonder if we can at least check that port 50000 is reachable?

Reply:
I can add a `telnet` command or something.

Reply:
Yep, or `nc -z 127.0.0.1 50000` (if I haven't messed up with the flags).

Reply:
Yep, I did that intead.